src/pages/spawn/spawnHost/fields/userTagsField/UserTagRow.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useState } from "react";
+import styled from "@emotion/styled";
+import IconButton from "@leafygreen-ui/icon-button";
+import { Input } from "antd";
+import { v4 as uuid } from "uuid";
+import Icon from "components/icons/Icon";
+import { PlusButton } from "components/Spawn";
+import { InputLabel } from "components/styles";
+import { InstanceTag } from "gql/generated/types";
+
+interface UserTagRowProps {
+  tag?: InstanceTag;
+  onDelete?: (key: string) => void;
+  onUpdateTag?: (tag: InstanceTag, deleteKey?: string) => void;
+  isNewTag?: boolean;
+}
+export const UserTagRow: React.FC<UserTagRowProps> = ({
+  tag,
+  onDelete,
+  onUpdateTag,
+  isNewTag = false,
+}) => {
+  const [key, setKey] = useState(tag?.key);
+  const [value, setValue] = useState(tag?.value);
+  const [hasEditedTag, setHasEditedTag] = useState(false);
+  const [createNewTag, setCreateNewTag] = useState(false);
+
+  const tagId = uuid();
+  const updateKey = (k) => {
+    setHasEditedTag(true);
+    setKey(k);
+  };
+  const updateValue = (v) => {
+    setHasEditedTag(true);
+    setValue(v);
+  };
+
+  const updateTag = () => {
+    if (tag?.key !== key) {
+      onUpdateTag({ key, value }, tag?.key);
+    } else {
+      onUpdateTag({ key, value });
+    }
+    setHasEditedTag(false);
+  };
+  const createNewTagHandler = () => {
+    onUpdateTag({ key, value });
+    setCreateNewTag(false);
+    setKey("");
+    setValue("");
+  };
+
+  const shouldShowTagInput = createNewTag || !isNewTag;
+  return (
+    <FlexColumnContainer
+      data-cy={shouldShowTagInput ? "user-tag-row" : "add-tag-button-row"}
+    >
+      {shouldShowTagInput && (
+        <FlexContainer>
+          <FlexColumnContainer>
+            <Section>
+              <InputLabel htmlFor={`tag_key_${tagId}`}>Key</InputLabel>
+              <Input
+                id={`tag_key_${tagId}`}
+                value={key}
+                onChange={(e) => updateKey(e.target.value)}
+                data-cy="user-tag-key-field"
+              />
+            </Section>
+          </FlexColumnContainer>
+          <FlexColumnContainer>
+            <Section>
+              <InputLabel htmlFor={`tag_value_${tagId}`}>Value</InputLabel>
+              <Input
+                id={`tag_value_${tagId}`}
+                value={value}
+                onChange={(e) => updateValue(e.target.value)}
+                data-cy="user-tag-value-field"
+              />
+            </Section>
+          </FlexColumnContainer>
+
+          <IconButtonContainer>
+            {hasEditedTag ? (
+              <IconButton variant="light" aria-label="Edit Tag">
+                <Icon
+                  glyph="Edit"
```

Comment:
I don't understand this icon choice. This can display for new tags, which if you click it will add the tag. For that scenario, I think we want an icon that implies "confirm" or "accept," like a checkmark. For editing an existing tag, you can already edit it by typing in the box, so I would think we also want an icon that implies "confirm" don't we?

Reply:
Good idea